### PR TITLE
Ensure IP not empty in kubevirt_vmi_status_addresses metric

### LINKED ISSUE #13346: Controller metrics are broken without guest agent
**What happened**:
#12802 introduced a new metric that relies on the guest agent to fetch the IP addresses for each interface of a VMI, relying on the IP address to be the unique label value that sets the time-series apart.

Unfortunately, when any VMI has multiple interfaces but does not have a guest agent installed, the `address` label of the resulting metric is `""` for all of them, which is not a unique label. 

This causes the metrics engine to fail due to time-series collision: leader controller pod is returning this on `/metrics` endpoint with HTTP response code 500:
```
An error has occurred while serving metrics:

1 error(s) occurred:

* collected metric "kubevirt_vmi_status_addresses" { label:{name:"address" value:""} label:{name:"name" value:"<redacted>"} label:{name:"namespace" value:"<redacted>"} label:{name:"node" value:"<redacted>"} label:{name:"type" value:"InternalIP"} gauge:{value:1}} was collected before with the same name and label values
```

**What you expected to happen**:
The metrics are served properly.

**How to reproduce it (as minimally and precisely as possible)**:
- create a VM with 2 interfaces and not install/disable the guest agent
- port-forward the controller pods and check the `/metrics` endpoint on both of them

**Additional context**:
According to docs, the agent is optional, so having the agent installed should not be a requirement for having properly working controller metrics. Current behavior causes the `kubevirt_virt_controller_ready` metric to not include the broken controller and triggers the [LowVirtControllersCount](http://kubevirt.io/monitoring/runbooks/LowVirtControllersCount.html) alert.

**Environment**:
- KubeVirt version (use `virtctl version`): 1.4.0

### PR DESCRIPTION
**Pull request by machadovilaca**
### What this PR does
Before this PR:

Duplicate metrics with the same label values were created due to missing IP address in multus networks status, causing the Prometheus server to error and not expose any metrics.

After this PR:

Metric has an additional network name to avoid duplicates. There is also a fallback to not creating the metric if the network name and the IP address are not defined.

Fixes #13346

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
Ensure IP not empty in kubevirt_vmi_status_addresses metric
```

### CODE REVIEW

File: pkg/monitoring/metrics/virt-controller/vmistats_collector.go [hunk 1]
```diff
@@ -308,18 +308,24 @@ func collectVMIInterfacesInfo(vmi *k6tv1.VirtualMachineInstance) []operatormetri
...
+		}
 	}
 
 	return crs
 }
 
-func collectVMIInterfaceInfo(vmi *k6tv1.VirtualMachineInstance, iface k6tv1.VirtualMachineInstanceNetworkInterface) operatormetrics.CollectorResult {
-	return operatormetrics.CollectorResult{
+func collectVMIInterfaceInfo(vmi *k6tv1.VirtualMachineInstance, iface k6tv1.VirtualMachineInstanceNetworkInterface) *operatormetrics.CollectorResult {
+	if iface.IP == "" && iface.Name == "" {
```
Comment (avlitman) [inline pkg/monitoring/metrics/virt-controller/vmistats_collector.go:320]: what happen if just one of them is nil?

File: pkg/monitoring/metrics/virt-controller/vmistats_collector_test.go [hunk 2]
```diff
@@ -365,20 +365,52 @@ var _ = Describe("VMI Stats Collector", func() {
 			}
 		},
 			Entry("no interfaces", [][]string{}),
-			Entry("one interface", [][]string{{"192.168.1.2", "InternalIP"}}),
+			Entry("one interface", [][]string{{"default", "192.168.1.2", "InternalIP"}}),
```
Comment (avlitman) [inline pkg/monitoring/metrics/virt-controller/vmistats_collector_test.go:368]: default is not a bit confusing? what should be the default value here?

**Comment by avlitman** [comment]
this is what failed https://github.com/kubevirt/kubevirt/blob/main/tests/libnet/validation.go#L31 not sure why

**Review by jean-edouard — APPROVED** [review]

File: pkg/monitoring/metrics/virt-controller/vmistats_collector_test.go (hunk 2 above)
Comment (machadovilaca) [reply pkg/monitoring/metrics/virt-controller/vmistats_collector_test.go:368]: default is literally the usual value of the primary interface name, that's why I used it here

File: pkg/monitoring/metrics/virt-controller/vmistats_collector.go (hunk 1 above)
Comment (machadovilaca) [reply pkg/monitoring/metrics/virt-controller/vmistats_collector.go:320]: We create the metric with what is populated. This helps make sure the metrics have different label

File: pkg/monitoring/metrics/virt-controller/vmistats_collector_test.go (hunk 2 above)
Comment (avlitman) [reply pkg/monitoring/metrics/virt-controller/vmistats_collector_test.go:368]: +1 🙏🏼

**Comment by brianmcarey** [comment]
/override pull-kubevirt-e2e-k8s-1.31-sig-performance

Triggered by status-reconciler when lane was bumped.

**Pull request merged**